### PR TITLE
Stage 2 development : refactoring near 1.0.1 version updates. Fixing …

### PR DESCRIPTION
**Pull request by tiutiunnykleasoft**
…conflict while merging.

### CODE REVIEW

File: src/Service/Gateway.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
 use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
 use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Controller\ErrorController;
```
Comment (volgustleasoft) [inline src/Service/Gateway.php]: ErrorController is not needed in Gateway.php

File: src/Service/Helper.php [hunk 2]
```diff
@@ -68,61 +60,9 @@ class Helper
     private $loggerFactory;
 
     public function __construct(LoggerFactory $loggerFactory){
-        include (dirname(__FILE__)."/../Vendor/vendor/autoload.php");
```
Comment (volgustleasoft) [inline src/Service/Helper.php:71]: We need to include the autoloader.php once. Let's include it in the plugin boot file for the plugin installation.

Comment (tiutiunnykleasoft) [reply src/Service/Helper.php:71]: I can change to require_once.
But before refactoring, we included it every time for every helper class use.
Now it's changed to one time, in the moment of creating the ginger client.

File: src/Service/Gateway.php (hunk 1 above)
Comment (tiutiunnykleasoft) [reply src/Service/Gateway.php]: https://github.com/emspay/ems-online-shopware-6/commit/ad46c734d3e80c6b3224b676a128be6cb4326a6e

File: src/Service/Helper.php (hunk 2 above)
Comment (volgustleasoft) [reply src/Service/Helper.php:71]: Ok, let's replace the include with require_once

File: src/Service/Gateway.php (hunk 1 above)
Comment (volgustleasoft) [reply src/Service/Gateway.php]: There are still references to ErrorController in this file and in services.xml that need to be removed too.

**Review by volgustleasoft — COMMENTED** [review]
Condition logic does not match the logic of actions:
https://github.com/emspay/ems-online-shopware-6/blob/refactoring_1.0.1/src/Subscriber/paymentKeeper.php#L119

**Review by volgustleasoft — COMMENTED** [review]
Gateway.php
Let's get the client in the constructor, set it to the object property, and then use the property in the code so as not to get it twice through the builder. What do you think?

**Review by volgustleasoft — COMMENTED** [review]
Gateway.php
Remove mention AsyncPaymentProcessException and add EmsPluginException

**Review by volgustleasoft — COMMENTED** [review]
paymentKeeper.php
Replace localization check with billing address for country validation.

**Review by volgustleasoft — COMMENTED** [review]
There are unnecessary 'use' expressions in captureOrder.php and Webhook.php

**Review by volgustleasoft — COMMENTED** [review]
Add checks for the existence of plugin settings and if not, set them in the code ('BE,NL' and some just empty)

File: src/Service/Gateway.php (hunk 1 above)
Comment (tiutiunnykleasoft) [reply src/Service/Gateway.php]: @volgustleasoft done

File: src/Service/Helper.php (hunk 2 above)
Comment (tiutiunnykleasoft) [reply src/Service/Helper.php:71]: Done

**Comment by tiutiunnykleasoft** [comment]
> Gateway.php
> Let's get the client in the constructor, set it to the object property, and then use the property in the code so as not to get it twice through the builder. What do you think?

In that specific case, we must call the clientBuilder twice basically because for test API key logic we need to know what payment we use but the constructor of the Gateway.php calling much earlier before, at the moment where we don't know that information.

> Gateway.php
> Remove mention AsyncPaymentProcessException and add EmsPluginException

Done, thx.

> paymentKeeper.php
> Replace localization check with billing address for country validation.

Replaced : https://github.com/emspay/ems-online-shopware-6/pull/6/commits/f1bbba5c312ce8309d9152010bef42bfe9a81045

> There are unnecessary 'use' expressions in captureOrder.php and Webhook.php

Webhook.php use the necessary components which help to Route user

> There are unnecessary 'use' expressions in captureOrder.php and Webhook.php

Done.https://github.com/emspay/ems-online-shopware-6/pull/6/commits/5864ce35cc133c834881d9d74202b63f2e3c7e66

**Pull request merged**